src/data/protocols/db/guardian/save-token-repository.ts

```diff
@@ -1,7 +1,12 @@
 export interface SaveTokenRepository {
-  saveToken: (userId: string, token: string) => Promise<SaveTokenRepository.Result>
+  saveToken: (params: SaveTokenRepository.Params) => Promise<SaveTokenRepository.Result>
 }
 
 export namespace SaveTokenRepository {
+  export type Params = {
```

Comment:
the idea is to put them all in types or in interfaces. I opted for types because some parameters cannot be interfaces and it ended up being kind of strange to have type and interface to represent params and result.